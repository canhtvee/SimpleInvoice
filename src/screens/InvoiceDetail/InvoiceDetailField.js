import React from 'react';
import {View} from 'react-native';

import {AppText} from '../../commons';
import {Sizes} from '../../utils';

export function InvoiceDetailField({label, value}) {
  return (
    <View style={{flexDirection: 'row', marginVertical: Sizes.paddinglx}}>
      <AppText style={{width: Sizes.wpx(120)}}>{label}:</AppText>
      <AppText style={{overflow: 'hidden', width: Sizes.wpx(250)}}>
        {value}
      </AppText>
    </View>
  );
}
