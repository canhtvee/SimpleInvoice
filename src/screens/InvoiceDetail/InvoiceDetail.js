import {useRoute} from '@react-navigation/native';
import React from 'react';
import {ScrollView} from 'react-native';
import {AppContainer} from '../../commons';
import {Sizes} from '../../utils';
import {InvoiceDetailField} from './InvoiceDetailField';

export function InvoiceDetail() {
  const {params} = useRoute();

  const {invoice} = params;

  return (
    <AppContainer>
      <ScrollView
        contentContainerStyle={{
          paddingBottom: Sizes.padding * 2,
          paddingTop: Sizes.paddinglxx,
          alignItems: 'center',
        }}>
        {Object.keys(invoice).map((key, index) => (
          <InvoiceDetailField key={key} label={key} value={`${invoice[key]}`} />
        ))}
      </ScrollView>
    </AppContainer>
  );
}
