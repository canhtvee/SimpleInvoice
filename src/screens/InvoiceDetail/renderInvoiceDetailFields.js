import React from 'react';
import {InvoiceDetailField} from './InvoiceDetailField';

export function renderInvoiceDetailFields(data, rootPath) {
  return Object.keys(data).map(key => {
    if (!data[key]) {
      return null;
    }

    const _path = rootPath ? `${rootPath}.${key}` : key;

    if (typeof data[key] !== 'object') {
      return React.createElement(InvoiceDetailField, {
        key: _path,
        label: _path,
        value: `${data[key]}`,
      });
    }

    return renderInvoiceDetailFields(data[key], _path);
  });
}
