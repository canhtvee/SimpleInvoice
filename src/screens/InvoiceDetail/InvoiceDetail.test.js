import React from 'react';
import {cleanup, render} from '@testing-library/react-native';

import {InvoiceDetail} from './InvoiceDetail';

/**
 * Run test command
 * yarn test -u -t="InvoiceDetail"
 */

afterEach(cleanup);

jest.mock('@react-navigation/native', () => {
  return {
    useRoute: () => ({
      params: {invoice: {}},
    }),
  };
});

describe('InvoiceDetail', () => {
  test('should render correctly', () => {
    const tree = render(<InvoiceDetail />);

    expect(tree).toMatchSnapshot();
  });
});
