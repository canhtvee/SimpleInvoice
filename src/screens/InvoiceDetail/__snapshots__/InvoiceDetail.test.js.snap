// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvoiceDetail should render correctly 1`] = `
<RNCSafeAreaView
  edges={
    [
      "left",
      "right",
    ]
  }
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "justifyContent": "flex-start",
      },
      undefined,
    ]
  }
>
  <RCTScrollView
    contentContainerStyle={
      {
        "padding": 47,
      }
    }
  >
    <View />
  </RCTScrollView>
</RNCSafeAreaView>
`;
